crates/primitives/src/header.rs

```diff
@@ -206,9 +206,7 @@ impl Header {
     /// Heavy function that will calculate hash of data and will *not* save the change to metadata.
     /// Use [`Header::seal`], [`SealedHeader`] and unlock if you need hash to be persistent.
     pub fn hash_slow(&self) -> B256 {
-        let mut out = BytesMut::new();
-        self.encode(&mut out);
-        keccak256(&out)
+        keccak256(alloy_rlp::encode(self))
```

Comment:
I wanna keep self.encode but use vec instead, because then we could easily do pre allocate some capacity based on some heuristics